space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+![Space Aware Client](SAC.drawio.svg "Space Aware Client")
+The only requirement is that the space can be accessed through regular Kube APIs when using the appropriate kubeconfig information. 
+
+**Main features**  
+- Constantly watch for changes in the available spaces. The SAC is using an informer on Space objecs using the SF Space client-go APIs.
+    - Holds a cache of the pSpace access info for each Space 
+- Exposes utility functions to get RestConfig by simpy specifying the space name  
+
+Currently the SAC exposes a simple function that supply the config access info for a space. In the future we will add a SAC space aware ClientSets
+
+## Architecture details 
+When a new SpaceProviderDesc is created the SM creates a new namespace that will be used to host all the Space objects representing pSpaces on that space provider.  
+The name of the generated namespace will be `paceprovider-<provider name>`
+
+![Space, SpaceProviderDesc and the Space provider namespace](SF-NS.drawio.svg "SF Namespaces")
+
+Note: In the future the spaces will be created in their own namespace and not in namespace per provider. 
+
+The Space also holds a reference to the SpaceProviderDesc associated with this Space and some capabilities (e.g., the space type). The status of the space holds a secret reference to the secret holding the access information for that pSpace. As described before for managed spaces the SF retrive that information from teh space provider, create the needed secrets and update the Space object with the secret reference
+
+**Discovery**  
+As mentioned before, the Space manager also supports discovery of spaces created out of band (i.e., not through creating Space objects).
+
+Each provider adaptor constantly watch for pSpace creation/deletion events and feeds the SM with those events. In case a pSpace that matches the discovery filtering is detected and there is no corrresponding Space object the SM creates a Space object representing this new detected pSpace and sets its mode to “unmanaged” .
+
+**Cration & Deletion flow of managed Spaces**  
+The SM sets the status of the Space according to the actual staus of the pSpace. When a space is created a creation command is sent to the space provider and the state of the the Sapce is set to `Initializing`. Only when the pSpace is becoming avaliable (each space provider validate this in its own way) the SM sets the Space state to `Ready`.   
+The SM uses finalizers for the space deletion flow - the Space object is not removed until the corresponding pSpace is deleted by the space provider.
+
+## Examples
+
+**SpaceProviderDesc**
+The folowing is SpaceProviderDesc for for a KubeFlex provider type. The secret refernced by `secretRef` should be created before the SpaceProviderDesc object is created.
```

Comment:
(1) ".. is a SpaceProviderDesc .."

(2) What goes wrong if the Secret is not created first? FYI, Kubernetes style generally frowns on such ordering requirements.

Reply:
The SpaceProviderDesc will stay at Initializing and not move to Ready. Also an error message will be send.

Reply:
If it is allowed for the Secret to be created later and after that the relevant controller will react and proceed, that is the normal pattern and hardly needs comment in a high-level doc; certainly this is not a _requirement_ that the Secret be created first. If the only deviation from this pattern is that the controller logs an error message rather than a "waiting..." message while waiting, then I suggest fixing the logging to make a totally normal controller.